lib/widgets/reactive_widget.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class ReactiveWidget<T extends ChangeNotifier> extends StatefulWidget {
```

Comment:
What's the use case for this widget?

Reply:
Inspired by: https://www.filledstacks.com/post/flutter-provider-v3-architecture/#basewidget-implementation